TRACING-3289 |  Distributed tracing 2.9 tempo overview (#1)
max-cx/openshift-docs: modules/distr-tracing-architecture.adoc

** *Jaeger Console* – With the {JaegerName} user interface, you can visualize your distributed tracing data. On the Search page, you can find traces and explore details of the spans that make up an individual trace.

* *{TempoName}* - This component is based on the open source link:https://github.com/grafana/tempo/[Grafana Tempo project].

> 

Reviewer: Could you please clarify for each item:

`Gateway` there is one or (possibly) more than one?
`Distributor` there  is one or (possibly) more than one?
`Ingester` there is one or (possibly) more than one?
`Query Frontend` there is one or (possibly) more than one?
`Querier` there is one or (possibly) more than one?
`Compactor` there one or (possibly) more than one?

Author: Is it important? Tempo is a scalable system. All of these components can be scaled up.

Reviewer: It's not important at all. I just need to know whether to use `a` or `the`.

> ** *Gateway* – The Gateway handles authentication, authorization and forwards requests to the Distributor or Query frontend service.

Reviewer: ```suggestion
** *Gateway* – The Gateway handles authentication, authorization, and forwarding requests to the Distributor or Query front-end service.
```

> ** *Distributor* – The Distributor accepts spans in multiple formats including Jaeger, OpenTelemetry, Zipkin. It routes spans to ingesters by hashing the `+traceID+` and using a distributed consistent hash ring.

Reviewer: ```suggestion
** *Distributor* – The Distributor accepts spans in multiple formats including Jaeger, OpenTelemetry, and Zipkin. It routes spans to Ingesters by hashing the `+traceID+` and using a distributed consistent hash ring.
```

> ** *Ingester* – The Ingester batches trace into blocks, creates bloom filters and indexes, and then flushes it all to the backend. Blocks in the backend are generated in the following layout.

Reviewer: ```suggestion
** *Ingester* – The Ingester batches a trace into blocks, creates bloom filters and indexes, and then flushes it all to the back end. Blocks in the back end are generated in the following layout.
```
Also, as a reader, I could not find any `following layout`. Where exactly is it mentioned or shown?

Author: I am removing it it's from the grafana website

> ** *Query Frontend* – The Query Frontend is responsible for sharding the search space for an incoming query. The search query is then sent to the Queriers. The Query Frontend deployment exposes Jaeger UI through Tempo Query sidecar.

Reviewer: ```suggestion
** *Query Frontend* – The Query Frontend is responsible for sharding the search space for an incoming query. The search query is then sent to the Queriers. The Query Frontend deployment exposes the Jaeger UI through the Tempo Query sidecar.
```

> ** *Querier* - The Querier is responsible for finding the requested trace id in either the ingesters or the backend storage. Depending on parameters it will query both the ingesters and pull bloom/indexes from the backend to search blocks in object storage.

Reviewer: ```suggestion
** *Querier* - The Querier is responsible for finding the requested trace ID in either the Ingesters or the back-end storage. Depending on parameters, it can query the Ingesters and pull Bloom indexes from the back end to search blocks in object storage.
```
Also, `to search blocks in object storage` has which meaning?
1. To search for blocks (to find a specific block)
2. To search through blocks (to find something inside a block)

Author: It first searches blocks from the storage and then once they are loaded in the memory it searches through the blocks to find the data

Reviewer: Thank you for the explanation.

> ** *Compactor* – The Compactors stream blocks to and from the backend storage to reduce the total number of blocks.

Reviewer: ```suggestion
** *Compactor* – The Compactors stream blocks to and from the back-end storage to reduce the total number of blocks.
```
